crates/astria-composer/src/executor/bundle_factory/mod.rs

```diff
@@ -87,16 +87,21 @@ impl SizedBundle {
             return Err(SizedBundleError::SequenceActionTooLarge(seq_action));
         }
 
-        if self.curr_size + seq_action_size > self.max_size {
+        let new_size = self
+            .curr_size
+            .checked_add(seq_action_size)
+            .ok_or_else(|| SizedBundleError::NotEnoughSpace(seq_action.clone()))?;
+
+        if new_size > self.max_size {
             return Err(SizedBundleError::NotEnoughSpace(seq_action));
         }
 
         self.rollup_counts
             .entry(seq_action.rollup_id)
-            .and_modify(|count| *count += 1)
+            .and_modify(|count| *count = count.saturating_add(1))
```

Comment:
This one seemed highly unlikely to overflow to me, so I went with `saturating_add` mainly to avoid complicating the code, but I can change to return an error if anyone thinks it's a realistic possibility.